lib/msal-common/src/client/DeviceCodeClient.ts

```diff
@@ -147,62 +147,69 @@ export class DeviceCodeClient extends BaseClient {
-
-                        this.logger.error(`User defined timeout for device code polling reached. The timeout was set for ${userSpecifiedTimeout}`);
-                        clearInterval(intervalId);
-                        reject(ClientAuthError.createUserTimeoutReachedError());
-
-                    } else if (TimeUtils.nowSeconds() > deviceCodeExpirationTime) {
-
-                        if (userSpecifiedTimeout) {
-                            this.logger.verbose(`User specified timeout ignored as the device code has expired before the timeout elapsed. The user specified timeout was set for ${userSpecifiedTimeout}`);
-                        }
-
-                        this.logger.error(`Device code expired. Expiration time of device code was ${deviceCodeExpirationTime}`);
-                        clearInterval(intervalId);
-                        reject(ClientAuthError.createDeviceCodeExpiredError());
-
-                    } else {
-                        const thumbprint: RequestThumbprint = {
-                            clientId: this.config.authOptions.clientId,
-                            authority: request.authority,
-                            scopes: request.scopes
-                        };
-                        const response = await this.executePostToTokenEndpoint(
-                            this.authority.tokenEndpoint,
-                            requestBody,
-                            headers,
-                            thumbprint);
-
-                        if (response.body && response.body.error === Constants.AUTHORIZATION_PENDING) {
-                            // user authorization is pending. Sleep for polling interval and try again
-                            this.logger.info(response.body.error_description || "no_error_description");
-                        } else {
-                            clearInterval(intervalId);
-                            resolve(response.body);
-                        }
-                    }
-                } catch (error) {
-                    clearInterval(intervalId);
-                    reject(error);
-                }
-            }, pollingIntervalMilli);
-        });
+        while (continuePolling()) {
+            const thumbprint: RequestThumbprint = {
+                clientId: this.config.authOptions.clientId,
+                authority: request.authority,
+                scopes: request.scopes
+            };
+            const response = await this.executePostToTokenEndpoint(
+                this.authority.tokenEndpoint,
+                requestBody,
+                headers,
+                thumbprint);
+
+            if (response.body && response.body.error === Constants.AUTHORIZATION_PENDING) {
+                // user authorization is pending. Sleep for polling interval and try again
+                this.logger.info(response.body.error_description || "no_error_description");
```

Comment:
I would prefer something more meaningful here. Also, errors should be logged using `this.logger.error`.

Reply:
Replaced the `"no_error_description"` with `"Authorization pending. Continue polling."`, the response description is still most meaningful here as it is phrased like `"AADSTS70016: OAuth 2.0 device flow error. Authorization is pending. Continue polling. Trace ID: 01707a0c-640b-4049-8cbb-ee2304dc0700 Correlation ID: 78b0fdfc-dd0e-4dfb-b13a-d316333783f6 Timestamp: 2020-03-26 22:54:14Z"` with a lot of useful information for debugging.

Reply:
The message itself does not seem appropriate to categorize as an error message, I still think `this.logger.info` is the better categorization of the message.